File: C/Array/TwodArr.c
#include<stdio.h>
#include<stdbool.h>
bool search(int rows,int cols,int key,int matrix[rows][cols]){

    for(int i=0;i<rows;i++){
        for(int j=0;j<cols;j++){
            if(matrix[i][j]==key){
                printf("Key was found at : (%d,%d)",i,j);
                return true;
            }
        }
    }
    printf("Key not found");
    return false;
}
int main(){
    printf("Enter the size of rows\n");
    int rows;
    scanf("%d",&rows);
    
    printf("Enter the size of cols\n");
    int cols;
    scanf("%d",&cols);

    int arr[rows][cols];
printf("Enter the elements to the array ");
for(int i=0;i<rows;i++){
    for(int j=0;j<cols;j++){
        scanf("%d",&arr[i][j]);
    }
}
for(int i=0;i<rows;i++){
    for(int j=0;j<cols;j++){
        printf("%d",arr[i][j]);
        printf(" ");
    }
    printf("\n");
    }

    int key;
    printf("\nEnter the key to search\n");
    scanf("%d",&key);
    search(rows,cols,key,arr);
}

File: JavaProgLang/ArrayJava/TwodArr.java
import java.util.Scanner;

public class TwodArr {

    public static boolean search(int matrix[][],int key){
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]==key){
                    System.out.println("key was found at: ("+i+","+j+")");
                    return true;
                }
            }
        }
        return false;
    }
    public static void main(String args[]){
        int matrix[][]=new int[3][3];
            int n=matrix.length,m=matrix[0].length;

            Scanner sc=new Scanner(System.in);
            System.out.println("Enter elements to the matrix");
            for(int i=0;i<n;i++){
                for(int j=0;j<m;j++){
                    matrix[i][j]=sc.nextInt();
                }
            }
            for(int i=0;i<n;i++){
                for(int j=0;j<m;j++){
                    System.out.print(matrix[i][j]+" ");
                }
                System.out.println();
            }
            System.out.println("Enter the key to search");
                int key=sc.nextInt();
                boolean ans=search(matrix,key);     
    }
}
